File: src/main/java/com/pubnub/api/models/consumer/pubsub/PNMessageResult.java
package com.pubnub.api.models.consumer.pubsub;

import com.google.gson.JsonElement;

import lombok.ToString;

@ToString(callSuper = true)
public class PNMessageResult extends MessageResult {

    public PNMessageResult(BasePubSubResult basePubSubResult, JsonElement message) {
        super(basePubSubResult, message);
    }
}

File: src/main/kotlin/com/pubnub/api/models/consumer/pubsub/PNMessageResult.kt
package com.pubnub.api.models.consumer.pubsub

import com.google.gson.JsonElement
import com.pubnub.api.PubNubError
import com.pubnub.api.callbacks.SubscribeCallback

/**
 * Wrapper around an actual message received in [SubscribeCallback.message].
 */
data class PNMessageResult internal constructor(
    private val basePubSubResult: PubSubResult,
    override val message: JsonElement,
    val error: PubNubError? = null
) : MessageResult, PubSubResult by basePubSubResult
